shared/src/main/java/com/couchbase/lite/PropertiesImpl.java

```diff
@@ -58,26 +64,65 @@ public Object get(String key) {
 
     @Override
     public String getString(String key) {
-        if (properties != null)
-            return (String) properties.get(key);
-        else
-            return root == null ? null : root.get(key).asString();
+        if (properties != null) {
+            Object obj = properties.get(key);
+            if (obj != null && obj instanceof String)
+                return (String) obj;
+            else
+                return null;
+        } else {
+            if (root == null)
+                return null;
+            else {
+                FLValue flvalue = root.get(key);
+                if (flvalue.getType() == kFLString)
+                    return flvalue.asString();
+                else
+                    return null;
+            }
+        }
     }
 
     @Override
     public int getInt(String key) {
-        if (properties != null)
-            return (int) properties.get(key);
-        else
-            return root == null ? 0 : root.get(key).asInt();
+        if (properties != null) {
+            Object obj = properties.get(key);
+            if (obj != null && obj instanceof Number)
+                return ((Number) obj).intValue();
+            else
+                return 0;
+        } else {
+            if (root == null)
+                return 0;
+            else {
+                FLValue flvalue = root.get(key);
+                if (flvalue.getType() == kFLNumber)
+                    return flvalue.asInt();
+                else
+                    return 0;
+            }
+        }
     }
```

Comment:
Now I got your point.

In case of `asBool()` method, it returns true for any string value. 
But Java, Boolean.valueof(randam String) return false.
It is reason that I added type check before returning value.

In case of `asInt()` it returns always 0 for non number values.

```
    bool Value::asBool() const noexcept {
        switch (tag()) {
            case kSpecialTag:
                return tinyValue() == kSpecialValueTrue;
            case kShortIntTag:
            case kIntTag:
            case kFloatTag:
                return asInt() != 0;
            default:
                return true;
        }
    }
```